File: BPGezinswetenschappen.DAL/Models/Proposal.cs
namespace BPGezinswetenschappen.DAL.Models
{
    public class Proposal
    {
        // [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ProposalId { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public string? Origin { get; set; }


        // navigation property

        public ICollection<Project>? Projects { get; set; }
        public ICollection<Topic>? Topics { get; set; }


    }
}

File: Client/src/app/shared/models/proposal.ts
import { Topic } from './topic';
import { User } from './user';
import { Project } from './project';

export interface Proposal {
  proposalId: number;
  title: string;
  description: string;
  origin?: string;
  topics?: Topic[];
}

export interface ProposalCreateDto {
  title: string;
  description: string;
  origin?: string;
  topicIds?: number[];
}

export interface ProposalReadDto {
  proposalId: number;
  title: string;
  description: string;
  origin?: string;
  topics?: Topic[];
}

export interface ProposalUpdateDto {
  title?: string;
  description?: string;
  origin?: string;
  topicIds?: number[];
}
